### PR TITLE
feat: theming

### PR DESCRIPTION
**Pull request by gavmck**
Adds some docs on theming, renames skin to theme and adds 3 demo themes.

### CODE REVIEW

File: docs/Theming.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+<Source code={theme} language='css' />
+
+These can be overridden by setting the `:root` CSS variables to create a base theme, as well as creating theme classes which can wrap components or groups of components.
+
+This setup allows consuming apps to create an infinite number of themes to apply to components, as well as allowing those themes to be applied at different levels within a page or app, rather than requiring them to be global.
+
+## Examples
+
+Diamond UI optionally includes an example set of 3 basic themes: light, medium and dark. The theme tokens are set on the `:root` and then applied to the base theme via classes, such as `.theme-dark`.
```
Comment (DanWebb) [inline docs/Theming.stories.mdx:24]:
```suggestion
Diamond UI optionally includes an example set of 3 basic themes: light, medium and dark. The theme tokens are set on the `:root` and then applied to the base theme via classes, such as `.diamond-theme-dark`.
```

File: docs/Theming.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+### Cards using demo themes
+
+Here is an example of 3 cards wrapped in the demo theme classes, which then apply the theme tokens.
+
+<Canvas>
+  <diamond-grid wrap="wrap">
+    <diamond-grid-item small-mobile="12" tablet="4">
+      <div class="diamond-theme-light">
+        <diamond-card>
```
Comment (DanWebb) [inline docs/Theming.stories.mdx:38]: Question: Should diamond elements support `class` the same way normal HTML elements support it? Then we won't need the wrapper div to apply the class

**Review by DanWebb — APPROVED** [review]

File: docs/Theming.stories.mdx (hunk 2 above)
Comment (gavmck) [reply docs/Theming.stories.mdx:38]: I think they might already...

Comment (gavmck) [reply docs/Theming.stories.mdx]: They do!

**Pull request merged**